Calculate the Required Sample Size:
Imagine the current checkout process has a conversion rate of 5%, and the bakery’s team is confident the new process will boost this to 7%.
Use Python to calculate the required sample size per group (current vs. new checkout process) with an effect size of 0.2, a significance level of 0.05, and a desired power of 0.8.

Analyze the Impact of Effect Size:

The bakery’s head chef, always aiming for perfection, wonders what would happen if the effect size were different. Calculate the required sample size for effect sizes of 0.1, 0.2, 0.3, and 0.4.
Explain how the sample size requirements change as the effect size changes. Why does this happen?

Explain the Relationship:

Imagine you’re explaining this to the bakery’s team in a fun, easy-to-understand way. Why is it so important to balance effect size and sample size when planning an A/B test? Help them understand how this ensures they’re not wasting time or resources and how it helps them confidently make decisions that could increase their sweet sales.




In [14]:
import statsmodels.stats.api as sm
base_rate = 0.05
new_rate = 0.07
effect_size = sm.proportion_effectsize(base_rate, new_rate)
print(effect_size)
alpha = 0.05
power = 0.8
sample_size = sm.NormalIndPower().solve_power(effect_size=effect_size, power=power, alpha=alpha)
print(f"Required sample size per group (effect size 0.2): {int(sample_size)}")

#using normalindpower instead of TTestIndPower as it fits more in this scenario.
#i.e as this a conversion rate with a relatively high expected sample size

-0.08449984251812542
Required sample size per group (effect size 0.2): 2198


since effect size is small, we'd need a high sample size(2198) to confirm the theory

In [15]:
effect_sizes = [0.1, 0.2, 0.3, 0.4]
sample_sizes = []

for es in effect_sizes:
    sample_size = sm.NormalIndPower().solve_power(effect_size=es, power=power, alpha=alpha)
    sample_sizes.append(int(sample_size))
    print(f"Effect size: {es}, required sample size: {int(sample_size)}")

Effect size: 0.1, required sample size: 1569
Effect size: 0.2, required sample size: 392
Effect size: 0.3, required sample size: 174
Effect size: 0.4, required sample size: 98


As opposite to the original effect size, the higher the value - the lower required sample size as it's less likely to be attributed to just normal fluctuations.

When creating a test with a relatively small(compared to 10% or more) effect size, often results can be just effects of random events or just a standard diviation, the lower the value - the harder it's to identify the reason behind this change. In order to not make a decision that based on an occasional accurance when trying to confirm such a theory you need a high sample size, to make sure that the reason behind enhanced conversion is the change you made, rather than some natural event